### PR TITLE
Implement html_render() 

### PR DESCRIPTION
**Pull request by andysim3d**
- add "integration_test"
- fix minor regexp bug

### CODE REVIEW

File: src/render/html_render.py [hunk 1]
```diff
@@ -2,8 +2,60 @@
...
+        return "<code>{}</code>".format(paragraph.content())
+    if isinstance(paragraph, HeaderParagraph):
+        return "<h{0}>{1}<h{0}>".format(paragraph.level(), paragraph.content())
+
+    _format = ""
+    if isinstance(paragraph, TextParagraph):
+        _format = "<p>{}</p>"
+    if isinstance(paragraph, ListWrapper):
+        if paragraph.is_ordered():
+            _format = "<ol>\n{}</ol>"
```
Comment (jinglongwu) [inline src/render/html_render.py:26]: why do we specifically need `\n` here?

File: src/render/html_render.py [hunk 2]
```diff
@@ -2,8 +2,60 @@
...
+        else:
+            _format = "<ul>\n{}</ul>"
+    if isinstance(paragraph, QuoteParagraph):
+        _format = "<blockquote>{}</blockquote>"
+    rendered_content = []
+    for element in paragraph.children:
+        rendered_content.append(_render_a_element(element))
```
Comment (jinglongwu) [inline src/render/html_render.py:33]: this structure limits we cannot have embedded paragraph.

File: src/render/html_render.py [hunk 3]
```diff
@@ -2,8 +2,60 @@
...
+
+
+def _render_a_element(element):
+    _format = ""
+    if isinstance(element, TextBlock):
+        return element.content()
+    if isinstance(element, ImgBlock):
+        return "<img src={} alt={}/>".format(element.url(), element.content())
+    if isinstance(element, LinkBlock):
+        return "<a herf={} />".format(element.content())
```
Comment (jinglongwu) [inline src/render/html_render.py]: It seems Link should have both content and url 
https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links

File: src/render/html_render.py [hunk 4]
```diff
@@ -2,8 +2,60 @@
...
+    if isinstance(element, CodeBlock):
+        return "<code>{}</code>".format(element.content())
+
+    if isinstance(element, ListParagraph):
```
Comment (jinglongwu) [inline src/render/html_render.py:48]: we may think of rename this ListParagraph to ListBlock

File: src/render/html_render.py [hunk 5]
```diff
@@ -2,8 +2,60 @@
...
+    if isinstance(element, ItalicBlock):
+        _format = "<em>{}</em>"
+    if isinstance(element, StrikethroughBlock):
+        _format = "<del>{}</del>"
+
+    return _format.format("".join(
+        [_render_a_element(i) for i in element.children]))
+
 
 def _dfs(node):
```
Comment (jinglongwu) [inline src/render/html_render.py]: we can get rid of this

File: src/render/html_render.py (hunk 1 above)
Comment (andysim3d) [reply src/render/html_render.py:26]: for formatting... but its not really necessary here.

File: src/render/html_render.py (hunk 2 above)
Comment (andysim3d) [reply src/render/html_render.py:33]: we don't embed paragraph.

**Review by jinglongwu — APPROVED** [review]

**Pull request merged**